tests/e2e/encrypted.sh

```diff
@@ -0,0 +1,179 @@
+export kbs_svc_name="kbs"
+export kbs_ingress_name="kbs"
+export runtimeclass="kata-qemu"
+export encrypted_image="ghcr.io/confidential/nginx:latest"
+
+build_encryption_key() {
+	docker build -t unencrypted .
+	# Encryption key needs to be 32 byte sequence
+ 	head -c 32 /dev/urandom | openssl enc > "${key_file}"
+	# Encryption key needs to be provided as base 64 enconded string
+	key_b64="$(base64 < ${key_file})"
+	key_id="kbs://${key_path}"
+
+	# Image encryption logic
+	guest_components_repo="https://github.com/confidential-containers/guest-components.git"
+	guest_components_path="${script_dir}/guest-components"
+	if [ -d "${guest_components_path}" ]; then
+		echo "Guest components repo directory exists"
+	else
+		git clone "$guest_components_repo"
+	fi
+
+	pushd "${guest_components_path}"
+  		tag_name="coco-keyprovider"
+ 		docker build -t "${tag_name}" -f ./attestation-agent/docker/Dockerfile.keyprovider .
+   		mkdir -p oci/{input,output}
+   		skopeo copy docker-daemon:unencrypted:latest dir:./oci/input
+   		docker run -v "${PWD}/oci:/oci" "${tag_name}" /encrypt.sh -k "${key_b64}" -i "${key_id}" -s dir:/oci/input -d dir:/oci/output
+   		skopeo inspect dir:./oci/output | jq '.LayersData[0].Annotations["org.opencontainers.image.enc.keys.provider.attestation-agent"] | @base64d | fromjson'
+ 		skopeo copy dir:./oci/output "docker://${encrypted_image}"
+		skopeo inspect "docker://${encrypted_image}" | jq -r '.Digest'
+		# Verify layer is encrypted
+		skopeo inspect "docker://${encrypted_image}" | jq -r '.LayersData[].MIMEType' | grep encrypted
+		skopeo inspect "docker://${encrypted_image}" | jq -r '.LayersData[].Annotations."org.opencontainers.image.enc.keys.provider.attestation-agent"' | base64 -d
+		skopeo inspect "docker://${encrypted_image}" | jq -r '.LayersData[].Annotations."org.opencontainers.image.enc.pubopts"' | base64 -d
+	popd
+}
+
+deploy_k8s_kbs() {
+	if [ ! -d "${script_dir}/trustee" ]; then
+		git clone "${trustee_repo}"
+	fi
+
+	pushd "${script_dir}/trustee/kbs/config/kubernetes"
+		echo "somesecret" > overlays/$(uname -m)/key.bin
+		export DEPLOYMENT_DIR=nodeport
+		./deploy-kbs.sh
+		kbs_pod=$(kubectl -n "${kbs_namespace}" get pods -o NAME)
+		kubectl wait --for=condition=Ready -n "${kbs_namespace}" "${kbs_pod}" --timeout=300s
+	popd
+}
+
+delete_k8s_kbs() {
+	pushd "${script_dir}/trustee/kbs/config/kubernetes"
+		kubectl delete -k overlays/$(uname -m)
+	popd
+	rm -rf "${script_dir}/trustee"
+}
+
+provide_image_key() {
+	kubectl exec -n "${kbs_namespace}" "${kbs_pod}" -- mkdir -p "/opt/confidential-containers/kbs/repository/$(dirname "$key_path")"
+	cat "${key_file}" | kubectl exec -i -n "${kbs_namespace}" "${kbs_pod}" -- tee "/opt/confidential-containers/kbs/repository/${key_path}" > /dev/null
+}
+
+launch_pod() {
+	kubectl apply -f "${script_dir}/nginx-encrypted.yaml"
+	export pod_name=$(kubectl -n default get pods -o NAME)
+}
+
+delete_pod() {
+	kubectl delete "${pod_name}"
+}
+
+check_image_key() {
```

Comment:
This is the part that currently is failing... seems like we can't find the secret...not sure if a configuration is missing @wainersm